src/linker/Linker/ConsoleLogger.cs

```diff
@@ -2,15 +2,59 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
+using System.Collections.Generic;
+using System.IO;
+using Mono.Cecil;
 
 namespace Mono.Linker
 {
 	public class ConsoleLogger : ILogger
 	{
-		public void LogMessage (MessageContainer message)
+		public Dictionary<MemberReference, string> ComputedStrings;
+
+		readonly List<MessageContainer> _messageContainers;
+		readonly MessageCategory _categoriesToCache;
+		readonly StreamWriter _streamWriter;
+
+		public ConsoleLogger (StreamWriter streamWriter, MessageCategory categoriesToCache = MessageCategory.None)
```

Comment:
Do we use the fact that we can redirect to any stream? If not, I would omit the parameter and simply do what we did before - `Console.WriteLine`.

Reply:
Also do we have a use case for caching anything but warnings?

I don't see a need to build a parametrized logger if we're not using it that way.

Reply:
> Also do we have a use case for caching anything but warnings?

I don't think so. I've changed this back so that we keep using `Console` and the only messages ever cached (and sorted) are the warnings.